### ISSUE TITLE
Cannot ask for more then one eviction set

### ISSUE THREAD

**Issue by van-ema** [cgvwzq/evsets#5]
Calling twice `find_evsets()` results in `num_evsets` to be equal to 2. 
However `get_evset(id=1)` return NULL . 

This can be solved initializing the variable `id` in `find_evsets` with the `num_evsets` value instead of 0 (line 289 of evets_api.c).

**Comment by cgvwzq** [comment]
Thanks for the feeedback. I already changed that, but right now the API is not consistent with anything, and probably it makes little sense to call find_evsets() more than once, since it starts all over again (you'll very likely find the same evset).

**Comment by cgvwzq** [comment]
Just for completeness.

The current approach to find more than one eviction set is using the flag `--findallcolors`. To avoid finding eviction sets for the same cache set, you can use `--findallcongruent`, which scans the whole list of addresses to identify which ones are evicted by the current eviction set (i.e., which other addresses are mapped into the same set), or even better (since it is faster), you can use `--conflictset`, which first generates a set of addresses that should have no more than associativity many addresses mapped into each cache set, and then find the different minimal eviction sets.

In summary to find all minimal eviction sets in your initial pool of addresses, run with flags `--findallcolors --conflictset`.
